### PR TITLE
Support resize data augmentation according to original image size 

### PR DESCRIPTION
**Pull request by Junjun2016**
Support resize data augmentation according to original image size (img_scale=None and ratio_range is tuple).

### CODE REVIEW

File: mmseg/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -18,7 +18,8 @@ class Resize(object):
     (multi-scale). There are 3 multiscale modes:
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:18]: Add `None` description.

File: mmseg/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -18,7 +18,8 @@ class Resize(object):
...
 
     - ``ratio_range is not None``: randomly sample a ratio from the ratio range
-    and multiply it with the image scale.
+    and multiply it with the image scale. When img_scale is None, img_scale is
+    the shape of image in results (img_scale = results['img'].shape[:2]).
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:22]:
```suggestion
    and multiply it with the image scale. When img_scale is None, the image is resized based on the original size. 
```

File: tests/test_data/test_test_time_aug.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import os.path as osp
```
Comment (xvjiarui) [inline tests/test_data/test_test_time_aug.py:1]: We may rename it to `test_tta.py`.

File: mmseg/datasets/pipelines/test_time_aug.py [hunk 4]
```diff
@@ -95,6 +102,11 @@ def __call__(self, results):
         """
 
         aug_data = []
+        if self.img_scale is None and mmcv.is_list_of(self.img_ratios, float):
+            img_scale = results['img'].shape[:2]
+            self.img_scale = [(int(img_scale[0] * ratio),
+                               int(img_scale[1] * ratio))
+                              for ratio in self.img_ratios]
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/test_time_aug.py:109]: `self.img_scale` is set after first `__call__`, so for the second image, it won't set again.

File: mmseg/datasets/pipelines/test_time_aug.py [hunk 5]
```diff
@@ -58,20 +58,27 @@ def __init__(self,
...
             img_ratios = img_ratios if isinstance(img_ratios,
                                                   list) else [img_ratios]
             assert mmcv.is_list_of(img_ratios, float)
-            assert isinstance(img_scale, tuple) and len(img_scale) == 2
+        if img_scale is None:
+            # mode 1: given img_scale=None and a range of image ratio
+            self.img_scale = None
+            assert mmcv.is_list_of(img_ratios, float)
+        elif isinstance(img_scale,
+                        tuple) and len(img_scale) == 2 and mmcv.is_list_of(
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/test_time_aug.py:69]: We move move `len(img_scale)` condition to assert.

Comment (Junjun2016) [reply mmseg/datasets/pipelines/test_time_aug.py]: Done!

File: mmseg/datasets/pipelines/test_time_aug.py (hunk 4 above)
Comment (Junjun2016) [reply mmseg/datasets/pipelines/test_time_aug.py]: fixed it and please check it again!

**Review by xvjiarui — APPROVED** [review]

File: mmseg/datasets/pipelines/test_time_aug.py (hunk 5 above)
Comment (xvjiarui) [reply mmseg/datasets/pipelines/test_time_aug.py]: `elif isinstance(img_scale, tuple) and mmcv.is_list_of(img_ratios, float):
assert len(img_scale) == 2 `

Comment (Junjun2016) [reply mmseg/datasets/pipelines/test_time_aug.py]: Fixed in the latest commit and please check it again.

**Comment by hellock** [comment]
Task linked: [CU-herb9d Resize data aug](https://app.clickup.com/t/herb9d)

**Pull request merged**